agent/agentssh/agentssh.go

```diff
@@ -515,8 +516,29 @@ func (s *Server) CreateCommand(ctx context.Context, script string, env []string)
 	if runtime.GOOS == "windows" {
 		caller = "/c"
 	}
+	name := shell
 	args := []string{caller, script}
 
+	if strings.HasPrefix(strings.TrimSpace(script), "#!") {
```

Comment:
I think this is acceptable due to the nature of terraform, but typically preceding space is not allowed for a shebang. A comment why we support it may be good though.

Reply:
I'll add a comment!